lib/pbench/agent/utils.py

```diff
@@ -390,3 +393,36 @@ def collect_local_info(pbench_bin):
         hostdata[arg] = cp.stdout.strip() if cp.stdout is not None else ""
 
     return (version, seqno, sha1, hostdata)
+
+
+class LocalRemoteHost:
+    """Dynamically build a list of all the local IP addresses so that a simple
+    simple query interface provides the answer to the question, "is the given
+    host name or IP address remote?"
+    """
+
+    def __init__(self):
+        """Build up a list of local IP addresses from all the interfaces.
+        """
+        local_ips_l = []
+        for ifaceName in interfaces():
+            for ifaddr in ifaddresses(ifaceName).setdefault(AF_INET, [{"addr": None}]):
+                ip_addr = ifaddr["addr"]
+                if ip_addr is None:
+                    continue
+                local_ips_l.append(ip_addr)
+        self.local_ips = frozenset(local_ips_l)
+
+    def is_remote(self, host_name):
+        """Returns True if the given host name (or IP address) is remote else
+        returns False.
+        """
+        try:
+            ip_addr = socket.gethostbyname(host_name)
+        except socket.gaierror:
```

Comment:
I don't know how expensive `gethostbyname()` is, but it might be worth a simple pre-filter:  if `host_name` doesn't start with an letter ([a-zA-Z]), then I don't think it can be a hostname; if it doesn't start with a hex-digit (or a colon or a left-square-bracket), then I don't think it can be an IP address; so we might be able to detect most of the error cases locally up front, and save the `try` blocks for the truly exceptional cases.

(You might also want to expand the function docstring to point out that errors result in `True` being returned.  Although, if we use a pre-filter to differentiate between names and addressses, we could just let the exceptions propagate and omit the `try` blocks.)